paperqa/settings.py

```diff
@@ -678,6 +678,19 @@ def _deprecated_field(self) -> Self:
                 setattr(self.agent.index, new_name, value)  # Propagate to new location
         return self
 
+    @model_validator(mode="after")
+    def _validate_temperature_for_o1_preview(self) -> Self:
```

Comment:
Can you modify one of our unit tests to use `o1` so this gets covered?

Reply:
Sure, should I put it in test_llms (already parameterized for gpt-4), test_configs (seems to be the central place for config testing), or some other file?

Reply:
Oh doesn't matter to me, I guess just pick one that seems to not rely on whatever model it was using previously. Defer to you here